apps/infrastructure/worker-api/app.py

```diff
@@ -1,41 +1,100 @@
+from apps.infrastructure.providers.provider import Provider
+from apps.infrastructure.utils import Config
 
 app = Flask(__name__)
-db.init_app(app)
+app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///workers.db"
+db = SQLAlchemy(app)
+
+states = {"creating": 0, "failed": 1, "success": 2, "destroyed": 3}
+
+
+class Worker(db.Model):
+    __tablename__ = "workers"
+
+    id = db.Column(db.Integer(), primary_key=True)
+    # user_id = db.Column(db.Integer())  # TODO: foreign key
+    provider = db.Column(db.String(64))
+    region = db.Column(db.String(64))
+    instance = db.Column(db.String(64))
+    state = db.Column(db.Integer)
+    created_at = db.Column(db.DateTime, default=datetime.now())
+    destroyed_at = db.Column(db.DateTime, default=datetime.now())
+
+    def __str__(self):
+        return f"<Worker id: {self.id}, Instance: {self.instance_type}>"
+
+    def as_dict(self):
+        return {c.name: getattr(self, c.name) for c in self.__table__.columns}
+
+
+def get_config(data):
+    """Reads environment variables from the domain instance to create a Config
+    object for deploying the worker."""
+    return Config(
+        app=Config(name="worker", count=1, id=db.session.query(Worker).count() + 1),
+        apps=[Config(name="worker", count=1)],
+        serverless=False,
+        websockets=False,
+        provider=os.environ["CLOUD_PROVIDER"],
+        vpc=Config(
+            region=os.environ["REGION"], instance_type=Config(**data["instance_type"])
+        ),
+    )
 
 
-@app.route("/workers", methods=["POST"])
+@app.route("/deploy", methods=["POST"])
 def create():
     """Creates a worker.
     This endpoint can be accessed by a user to create a new worker."""
 
-    data = json.loads(request.json)
-    config = Config(**data)
+    # data = json.loads(request.json)
+    config = get_config(request.json)
+    print(config)
 
+    deployment = None
     deployed = False
-    output = None
+    output = {}
 
     if config.provider == "aws":
-        aws_deployment = AWS_Serverfull(config=config)
-        deployed, output = aws_deployment.deploy()
+        deployment = AWS_Serverfull(config=config)
     elif config.provider == "azure":
         pass
     elif config.provider == "gcp":
         pass
 
-    if deployed:
-        # TODO: Use app.instance_type to pull worker specific data from cloud providers
-        # and fill them below
-        worker = Worker(user_id="", vCPU="", RAM="", storage="", GPU="", GPU_memory="")
+    if deployment.validate():
+        worker = Worker(
+            id=config.app.id,
+            provider=config.provider,
+            region=config.vpc.region,
+            instance=config.vpc.instance_type.InstanceType,
+            state=states["creating"],
+        )
         db.session.add(worker)
-        db.commit()
+        db.session.commit()
+
+        deployed, output = deployment.deploy()  # Deploy
+        # time.sleep(5)
+        # deployed = False
+        # output = {}
```

Comment:
Please remove this

```suggestion
        # time.sleep(5)
        # deployed = False
        # output = {}
```

Reply:
These are here for testing purpose, would be removed later, when we move the worker-api to domain api.